### DISCUSSION TITLE
Failed to create the second window

### DISCUSSION THREAD

**Discussion by xland** [mikke89/RmlUi#407] [Q&A] [answered]
Sorry to bother you again.

In my multi-window application, my first window can run normally.
But when I create the second window, an exception will be triggered.
Here is my code.

```c++
// application start 
// process messages continuously
void App::start() {
	while (windows.size()>0)
	{
		for (auto& win:windows)
		{
			win->ProcessEvents();
		}
	}
}
```

```c++
//WindowBase is the base class of all windows
void WindowBase::ProcessEvents()
{
	if (context_dimensions_dirty) {
		context_dimensions_dirty = false;
		Rml::Vector2i window_size;
		float dp_ratio = 1.f;
		glfwGetFramebufferSize(glfwWindow, &window_size.x, &window_size.y);
		glfwGetWindowContentScale(glfwWindow, &dp_ratio, nullptr);
		context->SetDimensions(window_size);
		context->SetDensityIndependentPixelRatio(dp_ratio);
	}
	glfwPollEvents();
	const bool result = glfwWindowShouldClose(glfwWindow);
	if (result) {		
		App::get()->closeWindow(this);
	}
	else {
		context->Update();
		renderInterface->BeginFrame();
		renderInterface->Clear();
		auto a = this->windowName;
		context->Render();
		renderInterface->EndFrame();
		glfwSwapBuffers(glfwWindow);
	}
}
```

The exception occurs in `RmlUi_Renderer_GL3.cpp` when I create a new window

```c++
void RenderInterface_GL3::RenderCompiledGeometry(Rml::CompiledGeometryHandle handle, const Rml::Vector2f& translation)
{
         //......
	//The code of RmlUi is omitted here. I haven't changed anything
	Gfx::CheckGLError("RenderCompiledGeometry");  //The exception occurred  here
}
```

The exception info is:

> 0xC0000005: An access violation occurred while reading position 0x00000000000000000.

The program is processing the first window's events when this exception occurred.

Can you tell me what's wrong with my code?

Thanks very much.

**Comment by xland** [comment]
Each window has its own `context` and `renderInterface`.
And every context and renderInterface are related.

```c++
context = Rml::CreateContext(windowName, Rml::Vector2i(width, height),renderInterface);
```
`Rml::SetRenderInterface` is not used to set global render interface

**Comment by xland** [comment, marked as answer]
This seems to be a silly question. I'd better turn it off temporarily

In `ProcessEvents` method must switch window context.

```c++
glfwMakeContextCurrent(glfwWindow);
```

**Comment by mikke89** [comment]
It seems like your applications is trying to read from a nullptr. It's a bit hard to help out in cases like this, it's really about debugging, figuring out which variable has the nullptr, and ensuring that everything is initialized in the correct order.

**Reply by xland** [reply]
Yes

In `ProcessEvents` method must switch window context.

```c++
glfwMakeContextCurrent(glfwWindow);
```
